File: Logic Building Problems/Basic Problems/Swap Two Numbers/Python/expected_approach.py
"""
Expected Approach Without using Third Variable
The idea is to swap two numbers using Arithmetic operators or Bitwise operators.
"""

if __name__ == "__main__":
    a: int = 2
    b: int = 3
    print(f"{a=}; {b=}")

    a, b = b, a

    print(f"{a=}; {b=}")

File: Logic Building Problems/Basic Problems/Swap Two Numbers/C++/expected_approach.cpp
"""
Expected Approach Without using Third Variable
The idea is to swap two numbers using Arithmetic operators or Bitwise operators.
"""

#include <iostream>

int main() {
    int a = 2, b = 3;

    printf("a = %d; b = %d\n", a, b);
  
    std::swap(a, b);
  
    printf("a = %d; b = %d", a, b);

    return 0;
}
